### PR TITLE
Mathieu/feat/list

### PR DESCRIPTION
**Pull request by MathieuRobilliard**

### CODE REVIEW

File: src/bl/listFacade.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                            return true
+                        } else {
+                            return false
+                        }
+                })
+            })
+        } catch (e) {
+            throw new ListNotFoundException(e)
+        }
+        return false
```
Comment (QuentinBouygues) [inline src/bl/listFacade.ts]: Could you inspire from the delete merged earlier ? Probably on card.

File: src/connectionParams.ts [hunk 2]
```diff
@@ -9,9 +9,9 @@ export const connectionOptions: MultipleConnections = {
         driver: {
             type: 'postgres',
             host: 'localhost',
-            port: 5434,
+            port: 5432,
```
Comment (QuentinBouygues) [inline src/connectionParams.ts]: This file can not be comitted

File: src/routes/list/list.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            }
+        } catch (e) {
+            res.status(404).json({ message: e.message})
+        }
+    }
+
+    static async getOneById(req: express.Request, res: express.Response) {
+        try {
+            var listId = req.params.list_id
+            if (listId == null) {
```
Comment (QuentinBouygues) [inline src/routes/list/list.ts]: No need for a var here, you can place `req.params.list_id` in the if statement

File: src/routes/list/list.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        } catch (e) {
+            res.status(404).json({ message: e.message})
+        }
+    }
+
+    static async insertFromBoardId(req: express.Request, res: express.Response) {
+        try {
+            var boardId = req.params.board_id
+            var listToInsert = req.body.list
+            if (boardId == null) {
```
Comment (QuentinBouygues) [inline src/routes/list/list.ts]: Same here, no need for a var

File: src/routes/list/list.ts [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+            }
+        } catch (e) {
+            res.status(404).json({ message: e.message})
+        }
+    }
+    
+    static async update(req: express.Request, res: express.Response) {
+        try {
+            var list = req.body.list
+            if (list == null) {
```
Comment (QuentinBouygues) [inline src/routes/list/list.ts]: Same here

File: src/routes/list/list.ts [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+            }
+        } catch (e) {
+            res.status(404).json({ message: e.message})
+        }
+    }
+
+    static async delete(req: express.Request, res: express.Response) {
+        try {
+            var listId = req.params.list_id
+            if (listId == null) {
```
Comment (QuentinBouygues) [inline src/routes/list/list.ts]: Same here

File: src/server.ts [hunk 7]
```diff
@@ -49,6 +50,13 @@ app.get('/protected', (req, res) => {
 
 app.post('/login', Login.authenticate)
 
+// ---------    List Routes   ---------
+app.get('/dashboards/:board_id/lists', List.getAllFromBoardId)
+app.get('/dashboards/:board_id/lists/:list_id', List.getOneById)
+app.put('/dashboards/:board_id/lists', List.insertFromBoardId)
```
Comment (QuentinBouygues) [inline src/server.ts]: Shouldn't this be a POST ?

**Review by QuentinBouygues — CHANGES_REQUESTED** [review]
This respects globally what have been done until now, only a few requests.

File: src/routes/list/list.ts (hunk 3 above)
Comment (Porquepix) [reply src/routes/list/list.ts]: And in case you need to store something, please use `const`.

**Review by QuentinBouygues — APPROVED** [review]

**Pull request merged**